rs/ic_os/metrics_tool/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+// TODO: refactor/merge this with fstrim_tool and guestos_tool metrics functionality
+use std::fs::File;
+use std::io::{self, Write};
+use std::path::Path;
+
+// TODO: everything is floating point for now
+pub struct Metric {
+    name: String,
+    value: f64,
+    annotation: String,
+    labels: Vec<(String, String)>,
+}
+
+impl Metric {
+    pub fn new(name: &str, value: f64) -> Self {
+        Self {
+            name: name.to_string(),
+            value,
+            annotation: "Custom metric".to_string(),
+            labels: Vec::new(),
+        }
+    }
+    pub fn with_annotation(name: &str, value: f64, annotation: &str) -> Self {
+        Self {
+            name: name.to_string(),
+            value,
+            annotation: annotation.to_string(),
+            labels: Vec::new(),
+        }
+    }
+
+    pub fn add_annotation(mut self, annotation: &str) -> Self {
+        self.annotation = annotation.to_string();
+        self
+    }
+
+    pub fn add_label(mut self, key: &str, value: &str) -> Self {
+        self.labels.push((key.to_string(), value.to_string()));
+        self
+    }
+
+    // TODO: formatting of floats
+    pub fn to_string(&self) -> String {
+        let labels_str = if self.labels.is_empty() {
+            String::new()
+        } else {
+            let labels: Vec<String> = self.labels.iter()
+                .map(|(k, v)| format!("{}=\"{}\"", k, v))
+                .collect();
+            format!("{{{}}}", labels.join(","))
+        };
+        format!("# HELP {} {}\n\
+             # TYPE {} counter\n\
+             {}{} {}", self.name, self.annotation, self.name, self.name, labels_str, self.value)
+    }
+}
+
+pub struct MetricsWriter {
+    file_path: String,
```

Comment:
Since it's a path, why not use the PathBuf type? Then you can remove String <-> Path/Buf conversions in other parts of the code.

Reply:
I would prefer to leave the API to use String for now as we will have to later refactor all similar APIs in other standalone tools as well.